aclocal.m4

```diff
@@ -20,106 +20,6 @@ You have another version of autoconf.  It may work, but is not guaranteed to.
 If you have problems, you may need to regenerate the build system entirely.
 To do so, use the procedure documented by the package, typically 'autoreconf'.])])
 
-# lt~obsolete.m4 -- aclocal satisfying obsolete definitions.    -*-Autoconf-*-
```

Comment:
I don't understand the question here - if there is one.

The removal of useless inlining of lt~obsolete.m4 in aclocal.m4 _is_ correct,
however @slouken insisted on keeping the version as was output on his own
setup.

Reply:
Oh sorry, I thought you were the one insisting on keeping these lines.
Running `./autogen.sh` on my sysstems keeps removing these lines.

Reply:
> Running `./autogen.sh` on my system keeps removing these lines.

And that's a good thing :)

Reply:
Okay, okay, I'll downgrade libtool so I don't keep adding those lines. :)

Reply:
> I'll downgrade libtool

What version do you have??? 2.4.7 is the official release already.
I think it may have something to do with the aclocal version in use,
though.

Reply:
I have automake 1.16.5 and autoconf 2.71 and I uninstalled libtool and now lt~obsolete.m4 is gone.

Reply:
When rebasing this pr and re-running `./autogen.sh`, `lt~obsolete.m4` appeared again.
I think this is caused by [this file](https://github.com/libsdl-org/SDL_image/blob/ec80a3aa0f10f6089c539381f41deef78cd2638f/acinclude/lt%7Eobsolete.m4#L1).
When I removed this file, `lt~obsolete.m4` disappears from `aclocal.m4` again.

Reply:
No that's incorrect.
I don't know what's happening with autoconf...

Reply:
It's okay again.
The `lt~obsolete.m4` version of current master is "serial 4", while libtool 2.4.7 has "serial 5".
This caused my libtool to copy its own `lt~obsolete.m4` into `aclocal.m4` instead of using the one committed into this repo.

Reply:
This will be the last message of my monologue and series of spam messages ;)
It looked like other libtool related m4 files were outdated as well, so I took the opportunity to update these as well.